ci: update BlueBuild, use BlueBuild action for PR builds

HastD opened the pull request:
* Update BlueBuild CLI to v0.9.27 (which fixes the bug that recently led to the BlueBuild CLI not being installed into the built images).
* Update BlueBuild action to v1.10.1.
* Use `maximize_build_space` and `verify_install` options for BlueBuild action now that these are added/improved in the latest version.
* Use BlueBuild action (rather than manually installing the CLI) for PR builds. This is made possible by the new version of the BlueBuild action having the option to opt out of pushing the image, and helps ensure that PR builds more closely match the real builds.

RoyalOughtness commented on .github/workflows/build-one-recipe.yml:
```diff
@@ -122,20 +112,21 @@ jobs:
           skip_checkout: true
           use_cache: false
           retry_push_count: 3
+          verify_install: true
```
we should also replicate this in the pr build (the signature verification)

RoyalOughtness commented on .github/workflows/build-one-recipe.yml:
```diff
@@ -122,20 +112,21 @@ jobs:
           registry_token: ${{ github.token }}
           pr_event_number: ${{ github.event.number }}
-          maximize_build_space: false
+          maximize_build_space: true
```
the pr build should also use your action

HastD committed:
ci: update BlueBuild, use BlueBuild action for PR builds

* Update BlueBuild CLI to v0.9.27 (which fixes the bug that recently led
  to the BlueBuild CLI not being installed into the built images).
* Update BlueBuild action to v1.10.1.
* Use `maximize_build_space` and `verify_install` options for BlueBuild
  action now that these are added/improved in the latest version.
* Use BlueBuild action (rather than manually installing the CLI) for PR
  builds. This is made possible by the new version of the BlueBuild
  action having the option to opt out of pushing the image, and helps
  ensure that PR builds more closely match the real builds.

HastD committed:
Merge branch 'live' into cli-version
